### PR TITLE
Local execution wav test

### PR DESCRIPTION
**Pull request by aitorrf**
- fix threads integration. 
- wav audio recorder
- clean main activity

### CODE REVIEW

File: library/src/main/java/com/voysis/recorder/WavAudioRecorder.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    private fun generateMimeType(): MimeType {
+        return MimeType(16000, 16, "signed-int", false, 1)
+    }
+
+    @Synchronized
+    override fun stop() {
+    }
+
+    private fun write(callback: OnDataResponse) {
+        if (wavFile == null) {
```
Comment (lduffy-voysis) [inline library/src/main/java/com/voysis/recorder/WavAudioRecorder.kt]: maybe we should check that the file exists also before proceeding

File: library/src/main/java/com/voysis/sevice/local/modules/VADService.kt [hunk 2]
```diff
@@ -106,19 +124,13 @@ internal class VADService(private val vadProcessor: Pointer,
             vadAudio.startFrame = cLib.get_vad_start_frame(data)
             vadAudio.stopFrame = cLib.get_vad_stop_frame(data)
         }
+        if (vadAudio.vadSignalled()) {
+            vadAudio.moreData = false
+        }
```
Comment (lduffy-voysis) [inline library/src/main/java/com/voysis/sevice/local/modules/VADService.kt]: whats the purpose of this moreData field?

Comment (aitorrf) [reply library/src/main/java/com/voysis/sevice/local/modules/VADService.kt]: moreData is one of the parameters of the native interface. This parameter indicate if there is more data to process. It is used in VAD and ASR. 
more data is false if vad is detected or if the client press stop and the VAD queue is empty

File: library/src/main/java/com/voysis/sevice/local/modules/VADService.kt [hunk 3]
```diff
@@ -56,10 +55,22 @@ internal class VADService(private val vadProcessor: Pointer,
 
     override fun <T> update(observable: GenericObservable, data: T) {
         if (data is ByteBuffer) {
-            execute(data)
+            execute(cloneByteBuffer(data))
         }
     }
 
+    private fun cloneByteBuffer(original: ByteBuffer): ByteBuffer {
```
Comment (oranoceallaigh) [inline library/src/main/java/com/voysis/sevice/local/modules/VADService.kt]: I'd like to have a design chat about the need for this change...it feels like the knowledge of where the audio is coming from is bleeding over into the rest of the code base.

Comment (aitorrf) [reply library/src/main/java/com/voysis/sevice/local/modules/VADService.kt]: of course, probably there any better way to do it

**Pull request merged**